src/main/java/io/github/slsa/models/Completeness.java

```diff
@@ -0,0 +1,65 @@
+package io.github.slsa.models;
+
+import java.util.Objects;
+
+/** Indicates that the builder claims certain fields in this message to be complete. */
+public class Completeness {
+
+  /**
+   * If true, the builder claims that recipe.arguments is complete, meaning that all external inputs
+   * are properly captured in recipe.
+   */
+  private boolean arguments;
+
+  /** If true, the builder claims that recipe.environment is claimed to be complete. */
+  private boolean environment;
+
+  /**
+   * If true, the builder claims that materials is complete, usually through some controls to
+   * prevent network access. Sometimes called “hermetic”.
+   */
+  private boolean materials;
+
+  public boolean isArguments() {
+    return arguments;
+  }
+
+  public void setArguments(boolean arguments) {
+    this.arguments = arguments;
+  }
+
+  public boolean isEnvironment() {
+    return environment;
+  }
+
+  public void setEnvironment(boolean environment) {
+    this.environment = environment;
+  }
+
+  public boolean isMaterials() {
+    return materials;
+  }
+
+  public void setMaterials(boolean materials) {
+    this.materials = materials;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Completeness that = (Completeness) o;
```

Comment:
I'm starting to see this override repeated a lot, would it be possible to genericise the type for that and then just move this to a higher class?

Reply:
As part of this change I'm just making sure that the legacy implementation works as is. I've only added the bare minimum to make my IDE not complain. A future change could move the Link to a more up to date format, removing gson and simplifying.